app/controllers/api/plans_base_controller.rb

```diff
@@ -29,40 +33,38 @@ def authorize_action
   end
 
   def resource(id = params[:id])
-    return unless id.present?
+    return if id.blank?
+
     collection.readonly(false).find(id)
   end
 
   def collection
-    raise UndefinedCollectionMethod.new('You have to override collection method')
-  end
-
-  def find_plan
-    @plan = resource
+    raise UndefinedCollectionMethod, 'You have to override collection method'
   end
 
-  def find_plans
-    search = ThreeScale::Search.new(params[:search] || params)
-    @plans = collection.not_custom
-                       .order_by(params[:sort].presence || :name, params[:direction].presence || :asc)
-                       .scope_search(search)
-    @page_plans = @plans.paginate(pagination_params)
+  def plan_type
+    raise UndefinedCollectionMethod, 'You have to override plan_type method'
   end
 
-  def find_issuer
-    @issuer = resource.issuer
+  def find_plan
+    @plan = resource
   end
 
   def find_service
     service_id = params[:service_id].presence || (@plan.issuer_id if @plan&.issuer_type == 'Service')
     return unless service_id
+
     @service = current_user.accessible_services.find(service_id)
     authorize! :show, @service
   end
 
   private
 
-  def create(attrs)
+  CREATE_PARAMS = %i[name system_name approval_required trial_period_days setup_fee cost_per_month].freeze
+  UPDATE_PARAMS = (CREATE_PARAMS - [:system_name]).freeze
+
+  def create
+    attrs = params.require(plan_type).permit(UPDATE_PARAMS)
```

Comment:
And yet tests passed, that is worrisome. Good catch!

Reply:
Actually, `system_name` is still passed in the next line in a weird way. This rings a bell... but I can't remember why it was done this way. It could be the result of a bad rebase so I'm gonna use CREATE_PARAMS and remove line 69.